### ISSUE TITLE
[OSX High SIerra] High CPU usage when Fn+Left & Fn+Right Macbook Pro

### ISSUE THREAD

**Issue by darkguy2008** [atom/atom#16276]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode 
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/ 
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Using High Sierra on a Macbook Pro, there is a huge delay when hitting Fn+Left or Fn+Right repeatedly. Even the initial keystroke is too slow, and CPU usage goes wild.

### Steps to Reproduce

1. Open Atom
2. Hit Fn+Left or Fn+Right
3. Fry an egg on your laptop due to the high CPU usage overheating your lappy.

**Expected behavior:** It should work as any other Home/End keystroke. Sublime text doesn't suffer from this. Keystroke should be immediate. This does not happen on Sierra though.

**Actual behavior:** High CPU usage and app hangs. Laptop gets hot. I'm not hungry, I don't wanna fry eggs yet.

**Reproduces how often:** Anytime, just hit those key combos repeatedly.

### Versions

You can get this information from copy and pasting the output of `atom --version` and `apm --version` from the command line. Also, please include the OS and what version of the OS you're running.

Both say not found (even though I ran the Install Shell Commands fuzzy finder script). OS High Sierra 10.13.1

### Additional Information

This also happens on VSCode ( bug here: https://github.com/Microsoft/vscode/issues/38975 ). Sublime Text or TextEdit don't suffer from this.

Example GIF:

![atom](https://user-images.githubusercontent.com/6304136/33236317-4fa6e198-d215-11e7-9b97-c7a63a86afd6.gif)

If there's anything else I can help with, I'll happily do. This is an important issue, as most of us are migrating to High Sierra and this makes both editors unusable.

Command line for helper processes are:

This one takes most of CPU (around 1.4% though):
```/Applications/Atom Beta.app/Contents/Frameworks/Atom Beta Helper.app/Contents/MacOS/Atom Beta Helper --type=renderer --enable-experimental-web-platform-features --no-sandbox --primordial-pipe-token=F55B20EF63092BAB0D162C13EB31095A --lang=en-US --app-path=/Applications/Atom Beta.app/Contents/Resources/app.asar --node-integration=true --disable-blink-features=Auxclick --hidden-page --enable-pinch --num-raster-threads=4 --enable-gpu-rasterization --enable-zero-copy --enable-gpu-memory-buffer-compositor-resources --enable-main-frame-before-activation --content-image-texture-target=0,0,3553;0,1,3553;0,2,3553;0,3,3553;0,4,3553;0,5,3553;0,6,3553;0,7,3553;0,8,3553;0,9,3553;0,10,34037;0,11,34037;0,12,34037;0,13,3553;0,14,3553;0,15,3553;1,0,3553;1,1,3553;1,2,3553;1,3,3553;1,4,3553;1,5,3553;1,6,3553;1,7,3553;1,8,3553;1,9,3553;1,10,34037;1,11,34037;1,12,34037;1,13,3553;1,14,3553;1,15,3553;2,0,3553;2,1,3553;2,2,3553;2,3,3553;2,4,3553;2,5,34037;2,6,3553;2,7,3553;2,8,3553;2,9,3553;2,10,3553;2,11,3553;2,12,34037;2,13,3553;2,14,34037;2,15,34037;3,0,3553;3,1,3553;3,2,3553;3,3,3553;3,4,3553;3,5,34037;3,6,3553;3,7,3553;3,8,3553;3,9,3553;3,10,3553;3,11,3553;3,12,34037;3,13,3553;3,14,34037;3,15,34037 --service-request-channel-token=F55B20EF63092BAB0D162C13EB31095A --renderer-client-id=4```

This one around 0.4% CPU:
```/Applications/Atom Beta.app/Contents/Frameworks/Atom Beta Helper.app/Contents/MacOS/Atom Beta Helper --type=gpu-process --no-sandbox --supports-dual-gpus=true --gpu-driver-bug-workarounds=0,1,10,23,25,35,38,45,51,59,61,62,63,64,66,70,71,73,81,82,83,86,88,89 --gpu-vendor-id=0x1002 --gpu-device-id=0x6821 --gpu-driver-vendor --gpu-driver-version --gpu-driver-date --gpu-secondary-vendor-ids=0x8086 --gpu-secondary-device-ids=0x0d26 --gpu-active-vendor-id=0x8086 --gpu-active-device-id=0x0d26 --service-request-channel-token=64584AC73A57378DC4F0C8589FA6BBDF```

Most CPU is taken by Atom Beta PID though (same with VSCode, and Brackets as well).

I'm also thinking that, if there's an easy way to have Atom in development running easily, I can run it, debug and try to see where the problem could be. What do you think?

**Comment by darkguy2008** [comment]
I'm referencing the issue here since it seems to happen with a simple Electron app:

https://github.com/electron/electron/issues/11262#issuecomment-346982359

**Comment by darkguy2008** [comment]
Nobody? :(

**Comment by 50Wliu** [comment]
> Nobody? :(

Please give us some time to respond :).  You reported this on a weekend right after a major US holiday and so lots of people were enjoying time off.

You mentioned that you installed Atom Beta - does the `atom-beta` command work in the Terminal?

**Comment by darkguy2008** [comment]
@50Wliu thanks for your answer! Indeed you are right, which makes sense. I'm sorry to be pushy, but it's making my everyday coding harder than usual :(

Okay, I managed to follow the missing steps, here's the rest of the info:

```
Atom    : 1.23.0-beta1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0
```

It can also be reproduced in safe mode. Actually, I found a way to reproduce this directly in a simple Electron app, so it happens in anything Electron-based including Atom.

Here's a GIF of what I've replicated. Using the keybinding package I can also notice that the delay exists before the package can get the key, so the problem is higher in the current chain of events:

![kbd1](https://user-images.githubusercontent.com/6304136/33289109-11d3f8ae-d384-11e7-9136-ade649d26427.gif)

I've also uploaded the CPU profile here: 
[CPU-20171127T150147.cpuprofile.zip](https://github.com/atom/atom/files/1507599/CPU-20171127T150147.cpuprofile.zip)

**Comment by 50Wliu** [comment]
@darkguy2008 thanks for the CPU profile and the related Electron issue.  The profile looks quite blank, which could make sense if it's something related with Electron's internals.  I'll be marking this as `electron` for now.

**Comment by darkguy2008** [comment]
Hi all, just wanted to mention that the problem seemed to be the Onyx app I ran a few days ago before I reported the bug ( https://www.titanium-software.fr/en/onyx.html ) **DON'T RUN IT IN HIGH SIERRA**. It will screw up your install and you'll have a BAD TIME (Undertale reference :P).

I fixed it by running CleanMyMac 3 and running all the maintenance scripts (ticked all checkboxes). Rebooted, and issue got fixed.

This can be closed :D (still, it's an interesting issue to check though). Thanks all for your time & input :)
